src/layer/utils.ts

```diff
@@ -33,55 +34,64 @@ export async function fetchGetCapabilitiesXml(
-    useCache: true,
-    ...getAxiosReqParams(reqConfig),
-  };
-  const queryString = stringify(query, { sort: false });
-  const url = `${baseUrl}?${queryString}`;
-  const res = await axios.get(url, axiosReqConfig);
-  const parsedXml = await parseStringPromise(res.data);
+  const parsedXml = await ensureTimeout(async innerReqConfig => {
+    const query = {
+      service: 'wms',
+      request: 'GetCapabilities',
+      format: 'text/xml',
+    };
+    const axiosReqConfig: AxiosRequestConfig = {
+      responseType: 'text',
+      useCache: true,
+      ...getAxiosReqParams(innerReqConfig),
+    };
+    const queryString = stringify(query, { sort: false });
+    const url = `${baseUrl}?${queryString}`;
+    const res = await axios.get(url, axiosReqConfig);
+    const parsedXml = await parseStringPromise(res.data);
+    return parsedXml;
+  }, reqConfig);
   return parsedXml;
 }
 
 export async function fetchGetCapabilitiesJson(
   baseUrl: string,
   reqConfig: RequestConfiguration,
 ): Promise<any[]> {
-  const query = {
-    request: 'GetCapabilities',
-    format: 'application/json',
-  };
-  const queryString = stringify(query, { sort: false });
-  const url = `${baseUrl}?${queryString}`;
-  const axiosReqConfig: AxiosRequestConfig = {
-    responseType: 'json',
-    useCache: true,
-    ...getAxiosReqParams(reqConfig),
-  };
-  const res = await axios.get(url, axiosReqConfig);
-  return res.data.layers;
+  const layers = await ensureTimeout(async innerReqConfig => {
+    const query = {
+      request: 'GetCapabilities',
+      format: 'application/json',
+    };
+    const queryString = stringify(query, { sort: false });
+    const url = `${baseUrl}?${queryString}`;
+    const axiosReqConfig: AxiosRequestConfig = {
+      responseType: 'json',
+      useCache: true,
+      ...getAxiosReqParams(innerReqConfig),
+    };
+    const res = await axios.get(url, axiosReqConfig);
+    return res.data.layers;
+  }, reqConfig);
+  return layers;
 }
 
 export async function fetchGetCapabilitiesJsonV1(
   baseUrl: string,
   reqConfig: RequestConfiguration,
 ): Promise<any[]> {
-  const instanceId = parseSHInstanceId(baseUrl);
-  const url = `https://eocloud.sentinel-hub.com/v1/config/instance/instance.${instanceId}?scope=ALL`;
-  const axiosReqConfig: AxiosRequestConfig = {
-    responseType: 'json',
-    useCache: true,
-    ...getAxiosReqParams(reqConfig),
-  };
-  const res = await axios.get(url, axiosReqConfig);
-  return res.data.layers;
+  const layers = await ensureTimeout(async innerReqConfig => {
+    const instanceId = parseSHInstanceId(baseUrl);
+    const url = `https://eocloud.sentinel-hub.com/v1/config/instance/instance.${instanceId}?scope=ALL`;
+    const axiosReqConfig: AxiosRequestConfig = {
+      responseType: 'json',
+      useCache: true,
+      ...getAxiosReqParams(innerReqConfig),
+    };
+    const res = await axios.get(url, axiosReqConfig);
+    return res.data.layers;
+  }, reqConfig);
+  return layers;
```

Comment:
I checked all the usages of `ensureTimeout`. We are now only wrapping public methods for classes exposed inside `index.ts` (no exposed functions required `ensureTimeout`).

It was an honest mistake on my part, I assumed that all the exported function with `reqConfig` are also available to users.

Reply:
Of course, no problem, I just hope it wasn't too much effort wasted... But it would be confusing if we left it like it was, it's much clearer now. Thank you for fixing this, and feel free to merge!